improver/blending/utilities.py

```diff
@@ -225,3 +227,76 @@ def _get_cycletime_point(cube: Cube, cycletime: str) -> int64:
         cycletime, time_unit=frt_units, calendar=frt_calendar
     )
     return round_close(cycletime_point, dtype=np.int64)
+
+
+def set_record_run_attr(
+    cubelist: CubeList, record_run_attr: str, model_id_attr: str
+) -> None:
+    """Set a model-cycle record attribute. From a list of cubes, model IDs and
+    cycle times are extracted. These are combined to create a shared record run
+    attribute that is applied to each cube in preparation for blending / combining
+    the cubes. The resulting blend will have a record of the latest cycle for
+    each model that contributed.
+
+    The cubes are modified in place.
+
+    Args:
+        cubelist:
+            Cubes from which to obtain model and cycle information, and to which
+            the resulting run record attribute is added.
+        record_run_attr:
+            The name of the record run attribute that is to be created.
+        model_id_attr:
+            The name of the attribute that contains the source model information.
+
+    Raises:
+        ValueError: If model_id_attr is not set and is required to construct a
+                    new record_run_attr.
+    """
+    if model_id_attr is None and not all(
```

Comment:
Why check `is None`? I don't think we want different behaviour if `model_id_attr = ""`
```suggestion
    if model_id_attr and not all(
```

Reply:
Then this needs to be `if not model_id_attr`, but yes, we can accommodate an empty string with this change.